### PR TITLE
kubernetes-helm: 2.11 -> 2.14.1

### PR DESCRIPTION
**Pull request by edude03**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by edude03** [comment]
@markus1189  @teozkr @nlewo @yurrriq, updates helm while still built from source, fixes the version issue raised in #56837

**Comment by mmahut** [comment]
@GrahamcOfBorg build kubernetes-helm

**Comment by teozkr** [comment]
That's a peculiar definition of friday! :D

**Comment by markus1189** [comment]
Nice!

File: pkgs/applications/networking/cluster/helm/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
 { stdenv, buildGoPackage, fetchFromGitHub }:
 
 buildGoPackage rec {
-  version = "2.11.0";
+  version = "2.14.1";
```
Comment (yurrriq) [inline pkgs/applications/networking/cluster/helm/default.nix]: I think we should do [2.14.2](https://github.com/helm/helm/releases/tag/v2.14.2)

**Review by yurrriq — CHANGES_REQUESTED** [review]

**Comment by yurrriq** [comment]
Any tips on how you got `dep init` and `vgo2nix` to work well here? In my experience it would seemingly hang for hours before I gave up.

**Comment by risicle** [comment]
`nox-review` builds, binaries execute on macos 10.13.

**Comment by andir** [comment]
This would also fix [CVE-2019-1000008](https://broken.sh/issues/CVE-2019-1000008). And if people using helm thinks it is alright also a backport.

**Pull request merged**